pkg/proxy/util/utils_test.go

```diff
@@ -537,3 +537,52 @@ func TestShuffleStrings(t *testing.T) {
 		}
 	}
 }
+
+func TestFilterIncorrectIPVersion(t *testing.T) {
+	testCases := []struct {
+		ipString        []string
+		isIPv6          bool
+		expectCorrect   []string
+		expectIncorrect []string
+	}{
+		{
+			ipString:        []string{"192.168.200.2", "192.1.34.23", "fd00:20::1", "2001:db9::3"},
+			isIPv6:          true,
+			expectCorrect:   []string{"fd00:20::1", "2001:db9::3"},
+			expectIncorrect: []string{"192.168.200.2", "192.1.34.23"},
+		},
+		{
+			ipString:        []string{"192.168.200.2", "192.1.34.23", "fd00:20::1", "2001:db9::3"},
+			expectCorrect:   []string{"192.168.200.2", "192.1.34.23"},
```

Comment:
I think it would be clearer to explicitly specify `isIPv6: false` on the ipv4 tests. (Also maybe something like `wantIPv6` or `expectIPv6` or `keepIPv6`?)

otherwise looks good

Reply:
yeah, also will add a desc field to describe the test

Reply:
done